code/modules/mob/living/carbon/human/human_update_icons.dm

```diff
@@ -288,140 +357,187 @@ There are several things that need to be remembered:
+		// SKYRAT EDIT ADDITION
+		else if(dna.species.bodytype & BODYTYPE_TESHARI)
+			icon_file = TESHARI_FEET_ICON
+		// SKYRAT EDIT END
+
+		if(!(icon_exists(icon_file, RESOLVE_ICON_STATE(worn_item))))
+			handled_by_bodytype = FALSE
+			icon_file = DEFAULT_SHOES_FILE
+
+		shoes_overlay = shoes.build_worn_icon(default_layer = SHOES_LAYER, default_icon_file = icon_file, override_file = handled_by_bodytype ? icon_file : null) // SKYRAT EDIT CHANGE
+
+		if(!shoes_overlay)
+			return
+		if(!(handled_by_bodytype) && (OFFSET_SHOES in dna.species.offset_features))
 			shoes_overlay.pixel_x += dna.species.offset_features[OFFSET_SHOES][1]
 			shoes_overlay.pixel_y += dna.species.offset_features[OFFSET_SHOES][2]
 		overlays_standing[SHOES_LAYER] = shoes_overlay
 
 	apply_overlay(SHOES_LAYER)
-*/
-//SKYRAT EDIT REMOVAL END
 
 
 /mob/living/carbon/human/update_inv_s_store()
 	remove_overlay(SUIT_STORE_LAYER)
 
 	if(client && hud_used)
 		var/atom/movable/screen/inventory/inv = hud_used.inv_slots[TOBITSHIFT(ITEM_SLOT_SUITSTORE) + 1]
-		inv.update_appearance()
+		inv.update_icon()
 
 	if(s_store)
-		s_store.screen_loc = ui_sstore1
-		if(client && hud_used?.hud_shown)
-			client.screen += s_store
-		update_observer_view(s_store)
-		overlays_standing[SUIT_STORE_LAYER] = s_store.build_worn_icon(default_layer = SUIT_STORE_LAYER, default_icon_file = 'icons/mob/clothing/belt_mirror.dmi')
-		var/mutable_appearance/s_store_overlay = overlays_standing[SUIT_STORE_LAYER]
+		var/obj/item/worn_item = s_store
+		var/mutable_appearance/s_store_overlay
+		update_hud_s_store(worn_item)
+
+		s_store_overlay = worn_item.build_worn_icon(default_layer = SUIT_STORE_LAYER, default_icon_file = 'icons/mob/clothing/belt_mirror.dmi')
+
+		if(!s_store_overlay)
+			return
 		if(OFFSET_S_STORE in dna.species.offset_features)
 			s_store_overlay.pixel_x += dna.species.offset_features[OFFSET_S_STORE][1]
 			s_store_overlay.pixel_y += dna.species.offset_features[OFFSET_S_STORE][2]
 		overlays_standing[SUIT_STORE_LAYER] = s_store_overlay
 	apply_overlay(SUIT_STORE_LAYER)
 
-
 /mob/living/carbon/human/update_inv_head()
-	..()
-	update_mutant_bodyparts()
-	var/mutable_appearance/head_overlay = overlays_standing[HEAD_LAYER]
-	if(head_overlay)
-		remove_overlay(HEAD_LAYER)
-		if(OFFSET_HEAD in dna.species.offset_features)
+	remove_overlay(HEAD_LAYER)
+	if(client && hud_used && hud_used.inv_slots[TOBITSHIFT(ITEM_SLOT_BACK) + 1])
+		var/atom/movable/screen/inventory/inv = hud_used.inv_slots[TOBITSHIFT(ITEM_SLOT_HEAD) + 1]
+		inv.update_icon()
+
+	if(head)
+		var/obj/item/worn_item = head
+		var/mutable_appearance/head_overlay
+		update_hud_head(worn_item)
+		var/handled_by_bodytype = FALSE
+		var/icon_file
+
+		// SKYRAT EDIT ADDITION - This needs to be refactored.
+		if((dna.species.bodytype & BODYTYPE_SNOUTED) && (worn_item.supports_variations_flags & CLOTHING_SNOUTED_VARIATION))
+			icon_file = worn_item.worn_icon_muzzled || SNOUTED_HEAD_FILE
+		else if(dna.species.bodytype & BODYTYPE_TESHARI)
```

Comment:
Can Tesharis have snouts?

Reply:
Not really, either way, it's a bitflag, their heads are style teshari